[IMP] website, website_blog, website_event: dynamic snippet improvements

anso-odoo opened the pull request:
This PR globally improves the `s_dynamic(_carousel)` snippet and reworks the different layouts of `s_blog_posts` and `s_events` snippets

**Improves the layout of the dynamic snippet:**

- Adds a title option that can be top, left or disabled
- Introduces a `data-column-classes` attribute for the dynamic snippet sub-templates that (if defined) will override the default dynamic snippet grid generation. Why ? Before this commit, we only had a mobile and a tablet and desktop layout (set by the `data-number-of-elements' attribute). Allowing custom classes is bringing more flexibility in terms of responsive layout when we don't need to use the default JS approach. 

**Improves the layout of the different templates of the blog post snippet: [1]**

- Adds an option to display/hide the author of the post
- Adds an option to switch between normal and smaller images (except for list layout that has no image)

_Big picture_
- Image is bigger by default
- Marley hover effect: a useless separation line was removed
- Dexter hover effect has been completely refactored
- Silly-chico hover effect has been improved (blog post titles are now
aligned with each other)

_Card layout_
- Card default design has been restored 
- Image is bigger by default
- An option has been added to show/hide the blog post teaser

_Horizontal layout_
- Is now a real card displayed horizontally
- This layout always show 1 blog post per row

_List layout_
- Added the date of the post
- Blog post picture has been switched with the author avatar (it didn't make sense to show the post picture in such a small format)
- Added the first lines of the blog

**Improves the layout of events snippet: [2]**

- Adapts the two templates to match the reworked blog ones 
- Adds an option to show/hide the time of the event (hours)

task-3097273

| Blog [1] | Before | After |
|--------|--------|--------|
| Big picture | <img width="428" alt="Screenshot 2023-12-19 at 11 39 14" src="https://github.com/odoo/odoo/assets/110090660/11f16eee-7ac3-42d0-bd8b-ee5192301370"> | <img width="442" alt="Screenshot 2023-12-19 at 11 40 35" src="https://github.com/odoo/odoo/assets/110090660/69eaae3b-82c9-4d44-b410-661a3b9502d0"> |
| Card | <img width="431" alt="Screenshot 2023-12-19 at 11 40 58" src="https://github.com/odoo/odoo/assets/110090660/2fd6a2e1-1306-4318-b16f-eeb72d9db81b"> |<img width="326" alt="Screenshot 2024-01-08 at 09 25 40" src="https://github.com/odoo/odoo/assets/110090660/0c4c2ca0-1ae7-4db1-82c2-c63956cb50f6"> |
| Horizontal | <img width="334" alt="Screenshot 2023-12-19 at 11 41 46" src="https://github.com/odoo/odoo/assets/110090660/d566cac4-2142-44ed-816e-b094a2d4a223"> | <img width="1321" alt="Screenshot 2024-01-08 at 09 26 40" src="https://github.com/odoo/odoo/assets/110090660/c6e2386c-04d3-460a-9e22-f64cb29bfb50"> |
| List | <img width="573" alt="Screenshot 2023-12-19 at 11 41 57" src="https://github.com/odoo/odoo/assets/110090660/d84c2824-09a7-435a-86e5-e2ef7f76485e"> | <img width="437" alt="Screenshot 2023-12-19 at 11 42 04" src="https://github.com/odoo/odoo/assets/110090660/08ca9eb7-aba7-4c65-a230-9b61812e2751"> | 

| Event [2] | Before | After |
|--------|--------|--------|
| Picture | <img width="317" alt="Screenshot 2023-12-19 at 11 46 30" src="https://github.com/odoo/odoo/assets/110090660/167fc2be-5a82-4e54-aedb-9f8fd67a3226"> | <img width="430" alt="Screenshot 2023-12-19 at 11 46 35" src="https://github.com/odoo/odoo/assets/110090660/b60caa8c-2b0a-4e9d-9531-39c658efb4d2"> |
| Card | <img width="323" alt="Screenshot 2023-12-19 at 11 46 56" src="https://github.com/odoo/odoo/assets/110090660/04c50e66-788d-46a2-946f-05a5be7482d3"> | <img width="325" alt="Screenshot 2023-12-19 at 11 46 46" src="https://github.com/odoo/odoo/assets/110090660/8414f28f-83db-4569-94a1-094814f42272"> | 

---
PR (ENT): https://github.com/odoo/enterprise/pull/68754
PR (DESIGN THEMES): https://github.com/odoo/design-themes/pull/897

I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/135509).

anso-odoo commented:
@stefanorigano Can you have a look at this please ? 👍

stefanorigano commented on addons/website_blog/static/src/snippets/s_blog_posts/000.scss:
```diff
@@ -1,64 +1,42 @@
             }
         }
+        .s_blog_posts_post {
+            height: 410px;
```
Please let's not hardcode layout values; it's a responsive environment with (soon) responsive font-size. 
I'd suggest to check https://getbootstrap.com/docs/5.1/helpers/ratio/#aspect-ratios

anso-odoo replied:
I tried to use these utility classes but there is an issue on larger screens when there is only 1 displayed element, the 1x1 ratio make the tiles look too big
<img width="1713" alt="Screenshot 2023-10-04 at 15 29 17" src="https://github.com/odoo/odoo/assets/110090660/e7934206-ca01-46f6-97a3-83fd11ea806f">

stefanorigano reviewed (changes requested):
Hello @anso-odoo ,
It looks good at first sight, thanks 👍 

Before jumping on the review  ~250 scss lines tho,  it would be better to fix the XML layout and overall edition behavior 😉

I've left some remarks

anso-odoo commented:
Hi @stefanorigano i removed the title option commit as we discussed (too many side effects and too complex to implement in a clean way) . I also simplified the animations that were the principal reason of the amount of scss lines in the file. Can you check the rest when you have time please ?

stefanorigano reviewed (commented):
Hei @anso-odoo ,
thanks so much...  but it will not be for 17.0 I guess 😢

There are still some details to fix in your branch, but the main issue is that whatever you design it will not look right on "intermediate" resolutions since the "large screen design" is triggered at the `770px` threshold, and that's obviously too low. 
![image](https://github.com/odoo/odoo/assets/8558351/c38da280-9e03-4bf2-81ae-6f52d7beb1e1)

The only solution I see is to rework the `DynamicSnippet` JS engine and we don't have time for that.
To add tons of ugly SCSS  overrides it's not an option, honestly I wouldn't feel to greenlight the approach.

The work you did so far wasn't wasted time tho, we'll use the designs you made to refactor `DynamicSnippet`accordingly 👍 

Follow some remarks on the code, just for your info.
I propose to keep this task/PR open (retargeting to master after the freeze).

Thanks again

anso-odoo commented on addons/website_blog/static/src/snippets/s_blog_posts/000.scss:
```diff
@@ -1,296 +1,147 @@
+        }
 
+        .s_blog_posts_post_cover {
+            max-height: 410px;
```
Thanks for the review ! Fixed the latest comment and I'm keeping this in the fridge for later 

Concerning this comment particularly: this piece of code is needed when the user decided to show only 1 article (https://github.com/odoo/odoo/pull/135509#discussion_r1345803201)

stefanorigano commented on addons/website/static/src/snippets/s_dynamic_snippet/000.js:
```diff
@@ -153,7 +150,7 @@ const DynamicSnippet = publicWidget.Widget.extend({
         const numberOfRecords = parseInt(dataset.numberOfRecords);
         let numberOfElements;
-        if (uiUtils.isSmall()) {
+        if (uiUtils.isLarge()) {
```
If `isLarge` then use the number of elements defined form mobile ? 🫤

stefanorigano reviewed (changes requested):
Hello @anso-odoo , thanks 👍 

Visual result is undoubtedly much better. There are some minor flaws, notably the difference between `LG` and `LG - 1px` that's way too "drastic" sometimes... an intermediate state at `md` would be welcome:

| `lg - 1px` | `lg` |
|--------|--------|
| <img width="920" alt="image" src="https://github.com/odoo/odoo/assets/8558351/b2d9fd70-3f81-4655-87b7-ae571e3d24f5"> | <img width="946" alt="image" src="https://github.com/odoo/odoo/assets/8558351/74cd3498-732f-40fd-951f-05732552b93e"> | 

Anyway, these are details. The main point is that in order to achieve these improvements you had to deviate even more than expected from the dynamic_snippet logic; in this last iteration you basically invalidated the "JS columns computation" while keeping there the code somehow.

This leads me thinking that we should probably review the specs and probably allow some templates to ignore the columns computation entirely (while keeping the fetch methods OFC). 

Let's take a moment to think and discuss about it ;)

stefanorigano reviewed (commented):
Hello @anso-odoo , well done, thanks 👍 

 I pushed some `--fixup` commits to streamline the remaining adaptations rather than going line-by-line with comments. Please take a moment to review them.

The visual result is mostly unchanged. Exceptions:
- the "silly-chico" line is rounded now  (It makes sense to me since the card is rounded as well)
- "compact" animation type for "Big Pictures"; if there is no room to show both title and excerpt on :hover, then we show the excerpt only.
- The "Image Size" option is now available for "Cards" and "Horizontal" layouts as well.

My main focus was make the branch work according to the user customizations and screen sizes.
The biggest challenge was the reduced available size when the title is on the left or when the container is small (or both 😅 ).

| Before (small + left) | After (small + left)|
|--------|--------|
| ![image](https://github.com/odoo/odoo/assets/8558351/ac39d91e-c700-4116-9317-a9da8882ecd0) |  ![image](https://github.com/odoo/odoo/assets/8558351/3076b0b7-49db-45ae-89fd-443320b300ca)|

Beside these, I've addressed several convention, typos and performance related issues. Notably:
- Use SCSS variables when possible and makes sense
- Avoid animating `width/height` properties
- Solve glitches by using `scale3d()` rather than `scale()` 
- Avoid `flex` layouts when possible.
- Avoid unnecessary complex html structures (I've unwrapped some elements from unnecessary `div`s, you'll get an huge diff for this reason)
- Avoid SCSS code repetition
- ...

I didn't check all tho, you may want to have a look at the rest of your code.

When ready, I'd suggest to re-check with @pierre-pde since there are several changes on the functional side.

I've briefed @Brieuc-brd to help you preparing the branch for the tech. review.

Thanks again and good luck 💪

Notes:
- you're adding trailing spaces, there are several vsCode extensions that highlight them 😉  
- I removed the "grid adaptation commit" out of this PR since it's unrelated
- You may want to rewrite your commits and PR's description
- I randomly experienced "infinite loading" and frozen UI while changing options ... not sure if it's related to your changes but if you can reproduce as well you should compare with master
- Carousels are still, well,  "problematic" by a layout standpoint. If you can't make it work we should probably disable "small container" and "title left" options for them... eventually check with pde first.

👋

anso-odoo commented:
@Brieuc-brd Thanks a lot for the review ! All your comments should be fixed now. 

> About image edition, I think they shouldn't be editable in the WB, could you check?

I guess we could ask to @pierre-pde what he thinks about that. Maybe the user will want to have a different image in the snippet and in the blog post ?

Thanks ! 💯

pierre-pde commented:
@anso-odoo , as pointed out by Brieuc, I don't think it's worth making it work to allow user choosing a different image for the snippet.
Let's remove the option to update/remove the picture.
I would do the following:
![image](https://github.com/odoo/odoo/assets/75257428/95ef29d3-50aa-4e39-b476-85083c41df24)

Brieuc-brd commented on addons/website_blog/views/snippets/s_blog_posts.xml:
```diff
@@ -4,103 +4,134 @@
-<template id="dynamic_filter_template_blog_post_list" name="List layout">
-    <div t-foreach="records" t-as="data" class="d-flex mt-3 s_blog_posts_post" data-number-of-elements="2">
+<template id="dynamic_filter_template_blog_post_list" name="List">
+    <div t-foreach="records" t-as="data" class="s_blog_posts_post p-3 border-0 rounded-3" data-extra-classes="g-3" data-column-classes="col-12 col-sm-6 col-lg-4">
```
`p-3` after border classes.

Brieuc-brd commented on addons/website_blog/views/snippets/s_blog_posts.xml:
```diff
@@ -4,103 +4,134 @@
+    <div t-foreach="records" t-as="data" class="s_blog_posts_post position-relative w-100" data-extra-classes="g-3" data-column-classes="col-12 col-sm-6 col-lg-4">
         <t t-set="record" t-value="data['_record']"/>
-        <a class="s_blog_posts_post_cover position-relative d-flex flex-column shadow-sm overflow-hidden rounded text-decoration-none" t-att-href="data['call_to_action_url']">
+        <a class="s_blog_posts_post_cover position-relative d-block h-100 shadow-sm overflow-hidden rounded text-decoration-none text-reset ratio ratio-1x1" t-att-title="'Read ' + data['name']" t-att-href="data['call_to_action_url']">
```
`rounded` after `h-100`

Brieuc-brd commented on addons/website_blog/views/snippets/s_blog_posts.xml:
```diff
@@ -4,103 +4,134 @@
+                    </div>
+                    <div class="s_blog_posts_post_cover_hidden h-auto py-2 px-4">
+                        <t t-call="website_blog.s_dynamic_snippet_template_author_placeholder"/>
+                        <p class="s_blog_posts_post_subtitle w-100 lead mt-2 m-0 overflow-hidden" t-field="record.subtitle"/>
```
`lead` after `m-0`

Brieuc-brd commented on addons/website_event/views/snippets/s_events.xml:
```diff
@@ -11,9 +11,9 @@
+    <figure t-foreach="records" t-as="data" class="s_events_event w-100" data-extra-classes="g-3" data-column-classes="col-12 col-sm-6 col-lg-4">
         <t t-set="record" t-value="data['_record']._set_tz_context()"/>
-        <a class="s_events_event_cover position-relative d-flex flex-column shadow-sm overflow-hidden rounded text-decoration-none"
+        <a class="s_events_event_cover position-relative d-flex align-items-center overflow-hidden rounded text-decoration-none ratio ratio-1x1"
```
`rounded` before `overflow-hidden`

Brieuc-brd reviewed (changes requested):
Thanks for your work @anso-odoo 👍 
Some notes below.

About `o_dynamic_title`, the section is no longer editable, cool 👍 . But it would be nice to be able to edit just the text... but I'm not sure that's feasible. To be checked with the developers probably.
![Capture d’écran 2023-12-28 à 16 13 30](https://github.com/odoo/odoo/assets/80679690/33f0007b-6705-43c3-9531-8015e18dd3e3)

anso-odoo commented:
Hello @pierre-pde 
Concerning the edition of the cover image and the filter option, i'm not quite sure on how to do that actually. Do you think you could create a dev task for that ?
Apart from that, if you could test the result a last time and tell me if you find any flaw, that would be great. After that this PR will be ready for tech review !
cc @stefanorigano

pierre-pde commented:
@anso-odoo , I'll create a task in RD website once this one is merged for the cover image/filter.

Regarding your branch, I some remarks, let me know if you prefer that RD website handles some of them.
1. would it be possible, for template big picture, card & list, to take full width even if number of fetched element is small?
![image](https://github.com/odoo/odoo/assets/75257428/e0b12910-491a-4ce6-b27e-3a4a617b8b56)
![image](https://github.com/odoo/odoo/assets/75257428/2cd4eda5-e575-4f76-8e86-d8dbfc48d5ad)

2. If the user deletes the title section:
a. ideally, the option in right panel should automatically switch to none.
b. if too hard, at least when changing the option back to none & then back to "Top" or "Left", the title should appear again.
Video; https://drive.google.com/file/d/12BODzl7ywD4Wto-RwXLJnouskMDBlrTT/view 

3. there is something a placeholder for text inside the block... we should avoid it.
![image](https://github.com/odoo/odoo/assets/75257428/488f252d-07e7-431b-a0fc-8257d866b2af)

anso-odoo commented:
Hello @pierre-pde 

> would it be possible, for template big picture, card & list, to take full width even if number of fetched element is small?

It's actually something we changed from the old design. Having a full width post doesn't really work well, instead for big pictures maybe... 

> If the user deletes the title section:

What about removing the delete option, like in the `s_table_of_content` snippet ? 

> there is something a placeholder for text inside the block... we should avoid it.

Should be fixed now :) 

Thanks again for your feedback !

stefanorigano reviewed (approved):
Thanks so much @anso-odoo , ready for tech. review 👍 

Notes:
- functional behavior approved by @pierre-pde during yesterday's onsite meeting.
- We'll create another snippet tailored for the "one blogpost only" scenario: the goal of this task is to to get a good visual result, regardless by the number of fetched element, snippet options and container sizes.

anso-odoo commented:
Hello @odoo/rd-website. Small up for this PR. Do you think someone in your team can have a look at it?

Thanks already !

qsm-odoo commented:
@anso-odoo It's still in my list for as soon as I get the time :+1:

rdeodoo commented:
FYI, on this opw the client wanted to hide the post date: https://www.odoo.com/web#id=3771992&cids=1&model=project.task&view_type=form

Maybe you could make it an option too? If I am not wrong by quickly testing you added a way to hide teaser and author but not the date

qsm-odoo commented on addons/website_blog/views/snippets/s_blog_posts.xml:
```diff
@@ -4,103 +4,142 @@
     </div>
 </template>
 
+<!-- Author ON / OFF -->
+<template id="website_blog.s_dynamic_snippet_template_author_placeholder" name="Author placeholder"/>
+<template id="website_blog.s_dynamic_snippet_template_author" inherit_id="website_blog.s_dynamic_snippet_template_author_placeholder" name="Author" active="True">
```
@anso-odoo Ok this is what you were explaining to me :) I think this is weird: do you agree or is this what you wanted? If this is not what you wanted, I can assign a dev to have a look at how it should be implemented (after you process the other comments :pray:).

anso-odoo replied:
Hello @qsm-odoo, thanks a lot for the review !

I agree that it's weird in a functional point of view. (For ref.: At this state of the PR, hiding/displaying these pieces of information has an impact on all the instances of `s_blog_post_snippet`s used across the website -> It shouldn't be the case.)

On a technical point of view, I did it this way this because in my mind it was cleaner than using something like `data-select-class="s_blog_post_author_hidden"` combined SCSS rules to `display: none` the information.

Anyway, I think it would be great if a dev could have a look indeed. I would be really happy to know the best solution for this !

Thanks !

qsm-odoo reviewed (changes requested):
Thanks for the work :+1: Here is a first batch of comments. Some things to adapt, some questions and a big question about the customize-views system. Process and answer what you can (try to push-force by adding the right adaptation to the right commits), then I'll have a deeper look (or another dev).

anso-odoo commented on addons/website_blog/static/src/snippets/s_blog_posts/000.scss:
```diff
@@ -1,296 +1,221 @@
-        font-weight: $headings-font-weight;
-        // Tweek line-height to help fit multi-line titles.
-        line-height: 1;
+    --Avatar-size: 1.5em;
```
Avatar is an already existing and shared component. In our [CSS variables naming guidelines](https://www.odoo.com/documentation/master/contributing/development/coding_guidelines.html#css-variables), we use uppercases for components.

stefanorigano commented on addons/website/static/src/snippets/s_dynamic_snippet/000.scss:
```diff
@@ -11,4 +11,42 @@
+            @include media-breakpoint-up(lg) {
+                // Switch to a fluid layout in order to keep entries readable.
+                // Our target grid is 2columns based
+                flex: 0 1 MAX($-entry-min-width, 50%);
```
This is uppercase on purpose in order to trigger the CSS function rather than the SASS one. 
(Sass is case sensitive, while css is not.)

_note:  Replying 'cause some changes (like this one) are mine, see https://github.com/odoo/odoo/pull/135509#pullrequestreview-1785499230_ 😉

qsm-odoo replied:
Oh ok, I would have imagined that the scss would be smarter than that and since those use different units (that the max SCSS function can thus not handle), it would fallback on the CSS function automatically (why the hell is this not the case? Are you sure it is not the case? :joy:). If not, I probably would have written it that way then:
```scss
flex: 0 1 max(#{$-entry-min-width}, 50%);
```
But if you don't agree, that's ok, this can stay this way.

stefanorigano replied:
> why the hell is this not the case? Are you sure it is not the case? 😂

Quite sure 😅
```console
"Internal Error: Incompatible units: 'px' and '%'.
```
Apparently saas introduced match function [way before CSS](https://sass-lang.com/documentation/values/calculations/#legacy-global-functions), and the back-compatibility logic doesn't cover the "incompatible units" scenario.

> But if you don't agree, that's ok, this can stay this way.

Yes, I'd like to keep it this way, thanks  👍

stefanorigano commented on addons/website/static/src/snippets/s_dynamic_snippet/000.scss:
```diff
@@ -11,4 +11,42 @@
+    > .o_container_small {
+        .s_dynamic_snippet_row > * {
+            @include media-breakpoint-up(lg) {
+                flex: 0 1 MAX($-entry-min-width, 50%);
```
> Is not this exactly the same as flex: 0 1 50% here ?

Yes, it's the case indeed 👍 
  
The goal **was** to handle potential user grid's customization, but this part of the code is triggered when `o_container_small` is applied only and `max()` will always return `50%` indeed.

stefanorigano commented on addons/website_blog/static/src/snippets/s_blog_posts/000.scss:
```diff
@@ -1,296 +1,221 @@
                 }
-                figcaption::after {
-                    transform-origin: bottom;
+                @include media-breakpoint-only(lg) {
+                    @include -big-picture-animate-compact();
                 }
-                .s_blog_posts_post_subtitle {
-                    opacity: 1;
-                    transform: translate3d(0, 0, 0);
+                @include media-breakpoint-up(xl) {
+                    @include -big-picture-animate-extended();
```
Yep, luckily there are not rules in `-big-picture-animate-extended()` that are **not** overridden  by `-big-picture-animate-compact()` when on `lg`, good catch 👍

qsm-odoo replied:
Oh I thought it would be the same in that case as well in fact. But apparently `@include media-breakpoint-down(lg) {` does not include LG so what you coded makes sense. They changed that in BS5 actually, I forgot. I don't find that "more logical" at all:
![image](https://github.com/odoo/odoo/assets/10338094/4e52c416-1fdf-4767-ae99-546c2ca7d1f2)

But indeed, if you override all properties, best goes with what I said to have less CSS rules.

anso-odoo commented on addons/website/views/snippets/snippets.xml:
```diff
@@ -593,7 +593,7 @@
         data-selector="section, section.s_carousel_wrapper .carousel-item"
         data-target="> *:has(> .row), > .s_allow_columns"
-        data-exclude=".s_masonry_block, .s_features_grid, .s_media_list, .s_showcase, .s_table_of_content, .s_process_steps, .s_image_gallery">
+        data-exclude=".s_masonry_block, .s_features_grid, .s_media_list, .s_showcase, .s_table_of_content, .s_process_steps, .s_image_gallery, .s_dynamic">
```
Not 100% sure but my guess is that before this PR the layout was not made in a way that it would activate the layout option ? The `.row` is now a direct child of the container, it was not the case before. Made a test and it looks like it comes from that.

Before:
<img width="247" alt="Screenshot 2024-05-07 at 14 24 16" src="https://github.com/odoo/odoo/assets/110090660/397c6d11-d134-4d2f-aaa1-8efbded465e7">

After this PR: 
<img width="290" alt="Screenshot 2024-05-07 at 14 28 27" src="https://github.com/odoo/odoo/assets/110090660/21bc979e-2950-4b14-8933-be3ae2069907">

qsm-odoo replied:
I see :+1:

qsm-odoo commented on addons/website_blog/static/src/snippets/s_blog_posts/000.scss:
```diff
@@ -1,296 +1,221 @@
+    --Avatar-size: 1.5em;
+
+    .s_blog_posts_post_teaser {
+        display: -webkit-box;
+        -webkit-line-clamp: 2;
+        -webkit-box-orient: vertical;
+
+        @include media-breakpoint-up(md) {
+            -webkit-line-clamp: 4;
+        }
```
Mmh, ok. As long as you guarantee it works on all major browsers :grimacing: And probably add a comment to explain the purpose of the 3 properties together... maybe create a mixin? I see you used it more than once.

anso-odoo commented:
Hello @qsm-odoo, thanks again for the review. I fixed what had to be fixed. 

I just have a question about https://github.com/odoo/odoo/pull/135509#discussion_r1584455819: I don't 100% get what you mean in this comment. IMO the current behavior of being able to modify and add content inside the title section is what we want ? Also the `_getContentEditableAreas` part is a bit shady for me. I'd be happy to test it but I'm not sure on how to proceed. 

Have a good day !

robinlej reviewed (commented):
Nice work, I'm sure this one granted a few headaches. :)

All in all, the dynamic snippets are much better and the custom-template-data feature works smoothly, well done. :clap:

anso-odoo commented on addons/website_blog/views/snippets/s_dynamic_snippet_blog_posts_preview_data.xml:
```diff
@@ -2,61 +2,50 @@
+            <div class="col-12 col-sm-6 col-lg-4">
+                <div class="s_blog_posts_post position-relative w-100">
+                    <a class="s_blog_posts_post_cover position-relative d-block h-100 rounded shadow-sm overflow-hidden text-decoration-none text-reset ratio ratio-1x1" href="#">
+                        <div class="o_record_cover_container d-flex flex-column h-100 o_colored_level o_cc w-100 bg-600 position-absolute o_dirty">
```
I cleaned this file as much as I could 👍

anso-odoo replied:
Edit: I restored the file before my cleaning but with your feedback integrated (also removed a useless `text-white` class). This file could indeed be decluttered but as it can be a tricky task (break css and stuff) I think it's safer to do it later 👍

anso-odoo commented on addons/website_event/static/src/snippets/s_events/000.scss:
```diff
@@ -20,17 +15,12 @@
 
             .s_events_event_cover {
-                min-height: 100%;
+                max-height: 410px;
```
We can indeed use rem instead of px in both places

Concerning a variable, I'm not sure, since the content of cards can differ, we might want to use different values 🤔

robinlej replied:
My concern was avoiding using the same or similar magic numbers in several places, but if design-wise we don't care that they differ :shrug:

robinlej commented on addons/website_blog/static/src/snippets/s_blog_posts/000.scss:
```diff
@@ -1,296 +1,225 @@
-            .s_blog_posts_post_title {
-                @include o-position-absolute(20px, 20px, auto, 20px);
+
+            .s_blog_posts_post_cover_hidden, .s_blog_post_big_picture_infos::before {
```
If you're ok with it, I am too :)

xO-Tx commented on addons/website/static/src/snippets/s_dynamic_snippet/000.scss:
```diff
@@ -1,14 +1,49 @@
 .s_dynamic {
-    &.o_dynamic_empty:not(.o_check_scroll_position) {
+    &.o_dynamic_snippet_empty:not(.o_check_scroll_position) {
```
> This rule causes an issue that wasn't visible before, but now that there are 2 columns in the dynamic snippet it is:

Hmmm... This is actually an existing issue linked to the rule before the current diff, you can simply reproduce it on master by setting the dynamic snippet's height to 100%:

https://github.com/user-attachments/assets/a79af1cb-b57e-4cbe-94f2-0b22ee29e2db

So I think it should be fixed in a separate PR, I'll add a task for it :+1:

bso-odoo commented:
@robodoo delegate=robinlej

robodoo commented:
Merge method set to rebase and fast-forward.

robodoo commented:
@anso-odoo @robinlej staging failed: ci/runbot on 40002c8bdfc144358607d83eabcc2bade1dfa3cd (view more at https://runbot.odoo.com/runbot/build/67649733)

robinlej commented:
@robodoo r+

robodoo commented:
@anso-odoo @robinlej linked pull request(s) odoo/enterprise#68754 not ready. Linked PRs are not staged until all of them are ready.

robodoo commented:
@anso-odoo @robinlej staging failed: ci/runbot on 2fa5c09e7da248767901ecbe8f2cdcd5c0a74726 (view more at https://runbot.odoo.com/runbot/build/67706383)

robinlej commented:
@robodoo r+
